hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionSystemContractModule.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec;
+
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.PrngSystemContract.PRNG_PRECOMPILE_ADDRESS;
+
+import com.hedera.node.app.service.contract.impl.annotations.TransactionScope;
+import com.hedera.node.app.service.contract.impl.exec.scope.HandleSystemContractOperations;
+import com.hedera.node.app.service.contract.impl.exec.scope.HederaOperations;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.PrngSystemContract;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Map;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.precompile.PrecompiledContract;
+
+@Module
+public interface TransactionSystemContractModule {
+    @Binds
+    @TransactionScope
+    HandleSystemContractOperations bindSystemContractOperations(
+            HandleSystemContractOperations handleSystemContractOperations);
+
+    @Provides
+    @TransactionScope
```

Comment:
I _suspect_ these can actually be `@Singleton` scope via the `@ProcessorModule`, because the `MessageFrame` passed into,
```
  PrecompileContractResult computePrecompile(
      final Bytes input, @Nonnull final MessageFrame messageFrame) {
    ...
```
will have a `ProxyWorldUpdater` that can do any `HederaOperations` work needed by the system contract.

Reply:
What about `HandleSystemContractOperations`?  Should I expose that as well via the `ProxyWorldUpdater`?  The only reason I have it at the moment is to create a child record but that will all change as discussed.  I'll leave it out for now until that api is clarified.